src/RoslynPad.Common.UI/ViewModels/MainViewModel.cs

```diff
@@ -432,6 +443,46 @@ public async Task OpenFile()
 
         OpenDocument(document);
     }
+    public async Task OpenFolder()
+    {
+        if (!IsInitialized) return;
+
+        var dialog = _serviceProvider.GetRequiredService<IOpenFolderDialog>();
+        //dialog.InitialDirectory = "C:\\Users\\luxus\\source\\repos\\roslynpad\\roslynpad\\src\\";
+        var folderPath = await dialog.ShowAsync().ConfigureAwait(true);
+
+        if (folderPath == null)
+        {
+            return;
+        }
+
+        if (!DocumentRoot.Path.Equals(folderPath, StringComparison.OrdinalIgnoreCase))
+        {
+            Settings.DocumentPath = folderPath;
+
+            DocumentRoot = SetDocumentRoot();
+        }
+
+        
+        var filePaths = Directory.GetFiles(folderPath, "*.cs, *.csx", SearchOption.TopDirectoryOnly);
+
+        foreach (var filePath in filePaths)
+        {
+            var normalizedPath = IOUtilities.NormalizeFilePath(filePath);
+            var document = DocumentViewModel.FromPath(normalizedPath);
+
+            if (!document.IsAutoSave)
+            {
+                var autoSavePath = document.GetAutoSavePath();
+                if (File.Exists(autoSavePath))
+                {
+                    document = DocumentViewModel.FromPath(autoSavePath);
+                }
+            }
+
+            OpenDocument(document);
```

Comment:
This bit of code is duplicated from another method in this class. Extract a common method and call from both